lightning/src/ln/channel.rs

```diff
@@ -192,6 +287,65 @@ enum OutboundHTLCState {
 	AwaitingRemovedRemoteRevoke(OutboundHTLCOutcome),
 }
 
+/// Exposes the state of pending outbound HTLCs.
+///
+/// Failure and success suffixes indicate the resolution of the HTLC.
+#[derive(Clone, Debug, PartialEq)]
+pub enum OutboundHTLCStateDetails {
+	/// THe AwaitingRemoteRevoke state indicates that this HTLC is not on any commitment transactions
+	/// yet, but we are awaiting the appropriate revoke_and_ack's from the remote before it will be
+	/// on the remote commitment transaction, possibly for a prior state first.
+	AwaitingRemoteRevokeToAdd,
+	/// The Committed state indicates that this HTLC is included on the remote's commitment
+	/// transaction. This includes the subsequent state where we include it in the local commitment
+	/// transaction, as:
```

Comment:
I trimmed this down. It was mainly copied from https://github.com/lightningdevkit/rust-lightning/blob/6b0ba8c9f154f9c0354ccf4d2d0f7fdd92470cbc/lightning/src/ln/channel.rs#L171-L182, but I agree this can be higher-level as it is user-facing documentation.